runtime/lua/vim/filetype.lua

```diff
@@ -23,36 +31,44 @@ local function starsetf(ft, opts)
 end
 
 ---@private
---- Get a single line or line range from the buffer.
---- If only start_lnum is specified, return a single line as a string.
---- If both start_lnum and end_lnum are omitted, return all lines from the buffer.
----
+--- Get a single line from the buffer.
+---@param bufnr integer The buffer to get the lines from
+---@param start_lnum integer The line number of the first line (inclusive, 1-based)
+---@return string
+function M.getline(bufnr, start_lnum)
+  -- Return a single line
+  return api.nvim_buf_get_lines(bufnr, start_lnum - 1, start_lnum, false)[1] or ''
+end
+
+---@private
+--- Get a line range from the buffer.
 ---@param bufnr integer The buffer to get the lines from
 ---@param start_lnum integer|nil The line number of the first line (inclusive, 1-based)
 ---@param end_lnum integer|nil The line number of the last line (inclusive, 1-based)
----@return table<string>|string Array of lines, or string when end_lnum is omitted
+---@return string[] # Array of lines
 function M.getlines(bufnr, start_lnum, end_lnum)
```

Comment:
Why is this on vim.filetype...? How do we not have enough "get lines" functions already? :)

Reply:
These are just convenience wrapper around the API which requires verbose usage. Marked private